app/src/sharedTest/java/org/oppia/app/activity/InputInteractionViewTestActivityTest.kt

```diff
@@ -71,14 +74,111 @@ class InputInteractionViewTestActivityTest {
     activityScenario.onActivity { activity ->
       activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
     }
-    onView(withId(R.id.test_number_input_interaction_view)).check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+    onView(withId(R.id.test_number_input_interaction_view)).check(matches(isDisplayed()))
+      .check(
+        matches(withText("9"))
+      )
+    activityScenario.recreate()
+    onView(withId(R.id.test_number_input_interaction_view)).check(matches(isDisplayed()))
+      .check(
+        matches(withText("9"))
+      )
+  }
+
+  fun testTextInputInteractionView_withNoInputText_hasCorrectPendingAnswerType() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_text_input_interaction_view) as TextInputInteractionView
+      assertThat(textAnswerRetriever.getPendingAnswer()).isInstanceOf(InteractionObject::class.java)
+      assertThat(textAnswerRetriever.getPendingAnswer().normalizedString).isEqualTo("")
+    }
+  }
+
+  @Test
+  fun testTextInputInteractionView_withInputtedText_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_text_input_interaction_view)).perform(typeText("abc"))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_text_input_interaction_view) as TextInputInteractionView
+      assertThat(textAnswerRetriever.getPendingAnswer()).isInstanceOf(InteractionObject::class.java)
+      assertThat(textAnswerRetriever.getPendingAnswer().objectTypeCase).isEqualTo(InteractionObject.ObjectTypeCase.NORMALIZED_STRING)
+      assertThat(textAnswerRetriever.getPendingAnswer().normalizedString).isEqualTo("abc")
+    }
+  }
+
+  @Test
+  fun testTextInputInteractionView_withInputtedText_onConfigurationChange_hasCorrectPendingAnswer() {
+    onView(withId(R.id.test_text_input_interaction_view)).perform(typeText("abc"))
+    activityScenario.onActivity { activity ->
+      activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+    }
+    onView(withId(R.id.test_text_input_interaction_view)).check(matches(isDisplayed()))
+      .check(
+        matches(withText("abc"))
+      )
+    activityScenario.recreate()
```

Comment:
Here & above: I found out that I was wrong about the requested orientation. This recreate() call and the onView() check below aren't needed. Please remove here & in other similar tests.

Reply:
removed